src/script/lua_api/l_util.cpp

```diff
@@ -358,10 +358,16 @@ int ModApiUtil::l_decompress(lua_State *L)
 	std::istringstream is(std::string(data, size), std::ios_base::binary);
 	std::ostringstream os(std::ios_base::binary);
 
-	if (method == LUA_COMPRESS_METHOD_DEFLATE) {
-		decompressZlib(is, os);
-	} else if (method == LUA_COMPRESS_METHOD_ZSTD) {
-		decompressZstd(is, os);
+	try {
+		if (method == LUA_COMPRESS_METHOD_DEFLATE) {
+			decompressZlib(is, os);
+		} else if (method == LUA_COMPRESS_METHOD_ZSTD) {
+			decompressZstd(is, os);
+		}
+	} catch (const SerializationError &e) {
+		lua_pushnil(L);
+		lua_pushstring(L, e.what());
+		return 2;
```

Comment:
switching from throw-on-error to return-on-error is a backwards incompatible change, which I'm afraid we can't do.

Reply:
Oh... I guess it will just have to be done in Lua with `pcall`. I'll adjust the PR to only be the `decode_base64` fix then.

Reply:
Fixed.